ccl/multiregionccl: TestRegionLivenessProberForLeases failed [slice bounds out of range]

ccl/multiregionccl.TestRegionLivenessProberForLeases [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18321801?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18321801?buildTab=artifacts#/) on master @ [d710b3b94c5c02fc6582d99057fe1296589b5082](https://github.com/cockroachdb/cockroach/commits/d710b3b94c5c02fc6582d99057fe1296589b5082):


```
github.com/cockroachdb/cockroach/pkg/server.(*Node).batchInternal(0x4006511508, {0x80efdc8?, 0x401dbecd28?}, {0x0?}, 0x4021fcb040)
	pkg/server/node.go:1651 +0x3e8
github.com/cockroachdb/cockroach/pkg/server.(*Node).Batch(0x4006511508, {0x80efdc8, 0x4012a56a00}, 0x4021fcb040)
	pkg/server/node.go:1852 +0x294
github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.func1({0x80efdc8?, 0x4012a56a00?}, {0x6752de0?, 0x4021fcb040?})
	pkg/rpc/context.go:713 +0x54
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.ServerInterceptor.func12({0x80efdc8, 0x4012a56a00}, {0x6752de0, 0x4021fcb040}, 0x401ad35460, 0x4053c17560)
	pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:96 +0x32c
github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.chainUnaryServerInterceptors.bindUnaryServerInterceptorToHandler.func4({0x80efdc8?, 0x4012a56a00?}, {0x6752de0?, 0x4021fcb040?})
	pkg/rpc/context.go:824 +0x3c
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func3({0x80efdc8, 0x4012a56a00}, {0x6752de0, 0x4021fcb040}, 0x40057da000?, 0x401ad35480)
	pkg/rpc/context.go:165 +0x88
github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.chainUnaryServerInterceptors.bindUnaryServerInterceptorToHandler.func4({0x80efdc8?, 0x4012a56a00?}, {0x6752de0?, 0x4021fcb040?})
	pkg/rpc/context.go:824 +0x3c
github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor({0x40057da000?, {{0x0?}, {0x81499a0?, 0x4052d463c0?}}}, {0x80efdc8, 0x4012a56a00}, {0x6752de0, 0x4021fcb040}, 0x401ad35460, 0x401ad354a0)
	pkg/rpc/auth.go:90 +0x15c
github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.chainUnaryServerInterceptors.bindUnaryServerInterceptorToHandler.func4({0x80efdc8?, 0x4012a56a00?}, {0x6752de0?, 0x4021fcb040?})
	pkg/rpc/context.go:824 +0x3c
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1.1({0x80efdc8?, 0x4012a56a00?})
	pkg/rpc/context.go:132 +0x40
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0x4032c1e480, {0x80efdc8, 0x4012a56a00}, {0x68ca3d3, 0x21}, 0x40076128d0)
	pkg/util/stop/stopper.go:349 +0x100
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1({0x80efdc8?, 0x4012a56a00?}, {0x6752de0?, 0x4021fcb040?}, 0x4007612958?, 0x1cf1e8c?)
	pkg/rpc/context.go:130 +0x7c
github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.chainUnaryServerInterceptors.bindUnaryServerInterceptorToHandler.func4({0x80efdc8?, 0x4012a56a00?}, {0x6752de0?, 0x4021fcb040?})
	pkg/rpc/context.go:824 +0x3c
github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.func2({0x80efdc8?, 0x4012a56a00?}, {0x40045b4500?, 0x42cbcc?}, {0x6752de0?, 0x4021fcb040?}, {0x65b1640, 0x401386e100}, 0x5cfe6a0?, {0xc679c00?, ...})
	pkg/rpc/context.go:723 +0x44
github.com/cockroachdb/cockroach/pkg/rpc.NewContext.ClientInterceptor.func8({0x80efdc8, 0x4012a56a00}, {0x68ca3d3, 0x21}, {0x6752de0, 0x4021fcb040}, {0x65b1640, 0x401386e100}, 0x0, 0x40340e8080, ...)
	pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:236 +0x2e8
github.com/cockroachdb/cockroach/pkg/rpc.getChainUnaryInvoker.func1({0x80efdc8, 0x4012a56a00}, {0x68ca3d3, 0x21}, {0x6752de0, 0x4021fcb040}, {0x65b1640, 0x401386e100}, 0x0, {0x0, ...})
	pkg/rpc/context.go:908 +0xdc
github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.func3({0x80efdc8, 0x4012a561e0}, 0x4021fcaf00, {0x0, 0x0, 0x0})
	pkg/rpc/context.go:793 +0x2a4
github.com/cockroachdb/cockroach/pkg/rpc.internalClientAdapter.Batch(...)
	pkg/rpc/context.go:916
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*grpcTransport).sendBatch(0x4045fffaa0, {0x80efdc8, 0x4012a561e0}, 0x1?, {0x80adb08, 0x400966f020}, 0x4021fcaf00)
	pkg/kv/kvclient/kvcoord/transport.go:209 +0x17c
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*grpcTransport).SendNext(0x4045fffaa0, {0x80efdc8, 0x4012a561e0}, 0x4021fcaf00)
	pkg/kv/kvclient/kvcoord/transport.go:187 +0x90
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendToReplicas(0x4004ac2608, {0x80efdc8, 0x4012a561e0}, 0x4021fca000?, {0x4034efe1e0, 0x402c660780}, 0x0)
	pkg/kv/kvclient/kvcoord/dist_sender.go:2755 +0x12d8
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendPartialBatch(0x4004ac2608, {0x80efdc8, 0x4012a561e0}, 0x4021fca000, {{0x403acb9098, 0x2, 0x2}, {0x4019eb7980, 0xb, 0x10}}, ...)
	pkg/kv/kvclient/kvcoord/dist_sender.go:2211 +0x518
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendPartialBatchAsync.func1({0x80efdc8?, 0x4012a561e0?})
	pkg/kv/kvclient/kvcoord/dist_sender.go:2070 +0x12c
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2({0x68e736c?, 0x402d137f98?})
	pkg/util/stop/stopper.go:498 +0x198
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 283598
	pkg/util/stop/stopper.go:488 +0x374
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegionLivenessProberForLeases.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-45872

ccl/multiregionccl.TestRegionLivenessProberForLeases [failed](https://tanzanite.cluster.engflow.com/invocations/default/c64576be-6cad-4dcd-8b43-75193d3a049f?testReportRun=20&testReportShard=3&testReportAttempt=1#targets-Ly9wa2cvY2NsL211bHRpcmVnaW9uY2NsOm11bHRpcmVnaW9uY2NsX3Rlc3Q=) on master @ [bc6d6e05a7c0f9ffd8103740239fdbc83fa78e3f](https://github.com/cockroachdb/cockroach/commits/bc6d6e05a7c0f9ffd8103740239fdbc83fa78e3f):

Fatal error:

```
panic: runtime error: slice bounds out of range [:3] with capacity 2 [recovered]
	panic: runtime error: slice bounds out of range [:3] with capacity 2 [recovered]
	panic: runtime error: slice bounds out of range [:3] with capacity 2 [recovered]
	panic: runtime error: slice bounds out of range [:3] with capacity 2

```
Stack: 

```
goroutine 314221 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0xc02690f120?, {0x89b9328, 0xc0267f4000})
	pkg/util/stop/stopper.go:226 +0x65
panic({0x6c8c6a0?, 0xc028849740?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0xc01060b240?, {0x89b9328, 0xc0267c0fa0})
	pkg/util/stop/stopper.go:226 +0x65
panic({0x6c8c6a0?, 0xc028849740?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).SendWithWriteBytes.func1()
	pkg/kv/kvserver/store_send.go:104 +0x148
panic({0x6c8c6a0?, 0xc028849740?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl_test.TestRegionLivenessProberForLeases.func3({0xc0267dca00?, 0x89b9328?}, 0xc02673db78?)
	pkg/ccl/multiregionccl/regionliveness_test.go:284 +0x146
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).SendWithWriteBytes(0xc009340008, {0x89b9328?, 0xc02673dae0?}, 0xc0267dca00)
	pkg/kv/kvserver/replica_send.go:161 +0x57e
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).SendWithWriteBytes(0xc008295808, {0x89b9328?, 0xc02673da48?}, 0xc0267dca00)
	pkg/kv/kvserver/store_send.go:188 +0x845
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Stores).SendWithWriteBytes(0xc01f766580, {0x89b9328, 0xc02673da48}, 0xc0267dca00)
	pkg/kv/kvserver/stores.go:198 +0x7e
github.com/cockroachdb/cockroach/pkg/server.(*Node).batchInternal(0xc008469508, {0x89b9328?, 0xc02673da48?}, {0xd433540?}, 0xc0267dca00)
	pkg/server/node.go:1651 +0x579
github.com/cockroachdb/cockroach/pkg/server.(*Node).Batch(0xc008469508, {0x89b9328, 0xc0267c0fa0}, 0xc0267dca00)
	pkg/server/node.go:1852 +0x278
github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.func1({0x89b9328?, 0xc0267c0fa0?}, {0x7010c00?, 0xc0267dca00?})
	pkg/rpc/context.go:713 +0x45
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.ServerInterceptor.func12({0x89b9328, 0xc0267c0fa0}, {0x7010c00, 0xc0267dca00}, 0xc03e25d620, 0xc00a173bc0)
	pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:96 +0x4b2
github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.chainUnaryServerInterceptors.bindUnaryServerInterceptorToHandler.func4({0x89b9328?, 0xc0267c0fa0?}, {0x7010c00?, 0xc0267dca00?})
	pkg/rpc/context.go:824 +0x34
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func3({0x89b9328, 0xc0267c0fa0}, {0x7010c00, 0xc0267dca00}, 0xc0267c0fa0?, 0xc03e25d640)
	pkg/rpc/context.go:165 +0x76
github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.chainUnaryServerInterceptors.bindUnaryServerInterceptorToHandler.func4({0x89b9328?, 0xc0267c0fa0?}, {0x7010c00?, 0xc0267dca00?})
	pkg/rpc/context.go:824 +0x34
github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor({0xc0037ba000?, {{0xc00f6347c0?}, {0x8a12ce0?, 0xc03aea7aa0?}}}, {0x89b9328, 0xc0267c0fa0}, {0x7010c00, 0xc0267dca00}, 0xc03e25d620, 0xc03e25d660)
	pkg/rpc/auth.go:90 +0x1e4
github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.chainUnaryServerInterceptors.bindUnaryServerInterceptorToHandler.func4({0x89b9328?, 0xc0267c0fa0?}, {0x7010c00?, 0xc0267dca00?})
	pkg/rpc/context.go:824 +0x34
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1.1({0x89b9328?, 0xc0267c0fa0?})
	pkg/rpc/context.go:132 +0x36
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0xc0435d1560, {0x89b9328, 0xc0267c0fa0}, {0x7187910, 0x21}, 0xc00f6348d0)
	pkg/util/stop/stopper.go:349 +0x15d
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1({0x89b9328?, 0xc0267c0fa0?}, {0x7010c00?, 0xc0267dca00?}, 0xc012809658?, 0xc1d65e2bb25fbbe3?)
	pkg/rpc/context.go:130 +0x8f
github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.chainUnaryServerInterceptors.bindUnaryServerInterceptorToHandler.func4({0x89b9328?, 0xc0267c0fa0?}, {0x7010c00?, 0xc0267dca00?})
	pkg/rpc/context.go:824 +0x34
github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.func2({0x89b9328?, 0xc0267c0fa0?}, {0x10?, 0xc021974780?}, {0x7010c00?, 0xc0267dca00?}, {0x6e6e720, 0xc0267c2700}, 0xc021974780?, {0x0, ...})
	pkg/rpc/context.go:723 +0x4c
github.com/cockroachdb/cockroach/pkg/rpc.NewContext.ClientInterceptor.func8({0x89b9328, 0xc0267c0fa0}, {0x7187910, 0x21}, {0x7010c00, 0xc0267dca00}, {0x6e6e720, 0xc0267c2700}, 0x0, 0xc033759900, ...)
	pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:236 +0x463
github.com/cockroachdb/cockroach/pkg/rpc.getChainUnaryInvoker.func1({0x89b9328, 0xc0267c0fa0}, {0x7187910, 0x21}, {0x7010c00, 0xc0267dca00}, {0x6e6e720, 0xc0267c2700}, 0x0, {0x0, ...})
	pkg/rpc/context.go:908 +0x13d
github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.func3({0x89b9328, 0xc0267f4000}, 0xc0267dc8c0, {0x0, 0x0, 0x0})
	pkg/rpc/context.go:793 +0x34f
github.com/cockroachdb/cockroach/pkg/rpc.internalClientAdapter.Batch(...)
	pkg/rpc/context.go:916
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*grpcTransport).sendBatch(0xc046cb3980, {0x89b9328, 0xc0267f4000}, 0x1?, {0x8977428, 0xc050e4c6c0}, 0xc0267dc8c0)
	pkg/kv/kvclient/kvcoord/transport.go:209 +0x19c
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*grpcTransport).SendNext(0xc046cb3980, {0x89b9328, 0xc0267f4000}, 0xc0267dc8c0)
	pkg/kv/kvclient/kvcoord/transport.go:187 +0x87
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendToReplicas(0xc00350e008, {0x89b9328, 0xc0267f4000}, 0xc0268db680?, {0xc01843ad20, 0xc0067032c0}, 0x0)
	pkg/kv/kvclient/kvcoord/dist_sender.go:2755 +0x1a09
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendPartialBatch(0xc00350e008, {0x89b9328, 0xc0267f4000}, 0xc0268db680, {{0xc0267d62a8, 0x2, 0x2}, {0xc026046420, 0xb, 0x10}}, ...)
	pkg/kv/kvclient/kvcoord/dist_sender.go:2211 +0x67b
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendPartialBatchAsync.func1({0x89b9328?, 0xc0267f4000?})
	pkg/kv/kvclient/kvcoord/dist_sender.go:2070 +0x11c
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2({0x71a4a42?, 0xc006d16fa0?})
	pkg/util/stop/stopper.go:498 +0x1f0
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 314213
	pkg/util/stop/stopper.go:488 +0x47c
```
<details><summary>Log preceding fatal error</summary>
<p>

```
*   | 	pkg/kv/kvserver/store_send.go:188
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Stores).SendWithWriteBytes
*   | 	pkg/kv/kvserver/stores.go:198
*   | github.com/cockroachdb/cockroach/pkg/server.(*Node).batchInternal
*   | 	pkg/server/node.go:1651
*   | github.com/cockroachdb/cockroach/pkg/server.(*Node).Batch
*   | 	pkg/server/node.go:1852
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.func1
*   | 	pkg/rpc/context.go:713
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.ServerInterceptor.func12
*   | 	pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:96
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.chainUnaryServerInterceptors.bindUnaryServerInterceptorToHandler.func4
*   | 	pkg/rpc/context.go:824
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func3
*   | 	pkg/rpc/context.go:165
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.chainUnaryServerInterceptors.bindUnaryServerInterceptorToHandler.func4
*   | 	pkg/rpc/context.go:824
*   | github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor
*   | 	pkg/rpc/auth.go:90
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.chainUnaryServerInterceptors.bindUnaryServerInterceptorToHandler.func4
*   | 	pkg/rpc/context.go:824
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1.1
*   | 	pkg/rpc/context.go:132
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
*   | 	pkg/util/stop/stopper.go:349
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1
*   | 	pkg/rpc/context.go:130
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.chainUnaryServerInterceptors.bindUnaryServerInterceptorToHandler.func4
*   | 	pkg/rpc/context.go:824
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.func2
*   | 	pkg/rpc/context.go:723
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewContext.ClientInterceptor.func8
*   | 	pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:236
*   | github.com/cockroachdb/cockroach/pkg/rpc.getChainUnaryInvoker.func1
*   | 	pkg/rpc/context.go:908
*   | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.func3
*   | 	pkg/rpc/context.go:793
*   | github.com/cockroachdb/cockroach/pkg/rpc.internalClientAdapter.Batch
*   | 	pkg/rpc/context.go:916
*   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*grpcTransport).sendBatch
*   | 	pkg/kv/kvclient/kvcoord/transport.go:209
*   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*grpcTransport).SendNext
*   | 	pkg/kv/kvclient/kvcoord/transport.go:187
*   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendToReplicas
*   | 	pkg/kv/kvclient/kvcoord/dist_sender.go:2755
*   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendPartialBatch
*   | 	pkg/kv/kvclient/kvcoord/dist_sender.go:2211
* Wraps: (2) runtime error: slice bounds out of range [:3] with capacity 2
* Error types: (1) *withstack.withStack (2) runtime.boundsError
*
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=20</code>
 - <code>shard=3</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegionLivenessProberForLeases.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/ccl/multiregionccl/regionliveness_test.go
+++ b/pkg/ccl/multiregionccl/regionliveness_test.go
@@ -281,7 +281,9 @@ func TestRegionLivenessProberForLeases(t *testing.T) {
 				keyToBlockMu.Lock()
 				keyPrefix := keyToBlock
 				keyToBlockMu.Unlock()
-				if keyPrefix == nil || !deleteRequest.Key[:len(keyPrefix)].Equal(keyPrefix) {
+				isPrefixToDelReq := len(deleteRequest.Key) >= len(keyPrefix) &&
+					!deleteRequest.Key[:len(keyPrefix)].Equal(keyPrefix)
+				if keyPrefix == nil || isPrefixToDelReq {
 					return nil
 				}
 				recoveryStart <- struct{}{}

